src/__spec_helper__/test-utils.js

```diff
@@ -98,6 +100,104 @@ const assertHoverTraversal = assertCorrectTraversal(wrapper => hoverList(wrapper
 
 const carbonThemesJestTable = carbonThemeList.map(theme => [theme.name, theme]);
 
+const spacingProps = [
+  ['m', 'margin'],
+  ['ml', 'marginLeft'],
+  ['mr', 'marginRight'],
+  ['mt', 'marginTop'],
+  ['mb', 'marginBottom'],
+  ['mx', 'marginLeft'],
+  ['mx', 'marginRight'],
+  ['my', 'marginTop'],
+  ['my', 'marginBottom'],
+  ['p', 'padding'],
+  ['pl', 'paddingLeft'],
+  ['pr', 'paddingRight'],
+  ['pt', 'paddingTop'],
+  ['pb', 'paddingBottom'],
+  ['px', 'paddingLeft'],
+  ['px', 'paddingRight'],
+  ['py', 'paddingTop'],
+  ['py', 'paddingBottom']
+];
+
+const getDefaultValue = (value) => {
+  if (typeof value === 'number') {
+    return `${value * 8}px`;
+  }
+  return value;
+};
+
+const testStyledSystemSpacing = (component, defaults, styleContainer) => {
+
+  describe('default props', () => {
+    const wrapper = mount(component());
+
+    it('should set the correct margins', () => {
+      let marginLeft;
+      let marginRight;
+      let marginTop;
+      let marginBottom;
+
+      if (defaults) {
+        marginLeft = getDefaultValue(defaults.ml || defaults.mx || defaults.m || undefined);
+        marginRight = getDefaultValue(defaults.mr || defaults.mx || defaults.m || undefined);
+        marginTop = getDefaultValue(defaults.mt || defaults.my || defaults.m || undefined);
+        marginBottom = getDefaultValue(defaults.mb || defaults.my || defaults.m || undefined);
+      }
+
+      expect(assertStyleMatch(
+        {
+          marginLeft,
+          marginRight,
+          marginTop,
+          marginBottom
+        },
+        styleContainer ? styleContainer() : wrapper
+      ));
+    });
+
+    it('should set the correct paddings', () => {
+      let paddingLeft;
+      let paddingRight;
+      let paddingTop;
+      let paddingBottom;
+
+      if (defaults) {
+        paddingLeft = getDefaultValue(defaults.pl || defaults.px || defaults.p || undefined);
```

Comment:
Just checking, is this the same order that takes presedence in styled-system?

Reply:
Yeah I checked it in a sandbox to make sure 👍